From 14a92cd3fa3e2f4b7b1f70a43b8d34e62c04b1f3 Mon Sep 17 00:00:00 2001
From: Yotam Ben Dosa <ybendosa@habana.ai>
Date: Mon, 14 Nov 2022 11:39:36 +0200
Subject: [PATCH] [SW-110273] gaudi2: add eROM TPM efuse cfg FIT

Add a target for creating a FIT that enables eROM TPM in EFUSE.

Change-Id: Ifb76626cc0a79454aacdd7a370aad95570443ea1
Signed-off-by: Yotam Ben Dosa <ybendosa@habana.ai>
---

diff --git a/scripts/make/gaudi2/Makefile b/scripts/make/gaudi2/Makefile
index 907abac..3cc76fc 100644
--- a/scripts/make/gaudi2/Makefile
+++ b/scripts/make/gaudi2/Makefile
@@ -125,7 +125,8 @@
 	${HL_DEV}-psu-ctrl.itb			\
 	${HL_DEV}-rma_erase.itb			\
 	${HL_DEV}-rma_status.itb		\
-	${HL_DEV}-secureboot_config.itb
+	${HL_DEV}-secureboot_config.itb		\
+	${HL_DEV}-erom_tpm_enable.itb
 
 .PHONY: gaudi2-security-fit-create
 gaudi2-security-fit-create: gaudi2-eeprom-read-create	\
@@ -133,7 +134,8 @@
 				gaudi2-update-svn-create gaudi2-fw-upd-closure-create	\
 				gaudi2-sram-sf-access-create gaudi2-psu-ctrl-create	\
 				gaudi2-rma-erase-create gaudi2-rma-status-create	\
-				gaudi2-secureboot-config-create
+				gaudi2-secureboot-config-create				\
+				gaudi2-erom-tpm-enable-create
 
 	@$(call hl-check-files,$(GAUDI2_SECURITY_FITS))
 	@tar czf $(IMAGES_DIR)/${HL_DEV}-sec-fit.tgz -C $(IMAGES_DIR)/ $(GAUDI2_SECURITY_FITS)
diff --git a/scripts/make/hl-common/Makefile b/scripts/make/hl-common/Makefile
index ba5ed14..5e87cf3 100644
--- a/scripts/make/hl-common/Makefile
+++ b/scripts/make/hl-common/Makefile
@@ -163,6 +163,11 @@
 	@$(if $(filter $(call hl-device, $@), $(HL_SEC_FIT_CREATE_ARCH)), \
 		${BR2_EXTERNAL}/board/habanalabs/common/create-simple.sh hl225b_enable fw-loader-conf no-use-elbi no-use-fwupdate load-bootloader)
 
+.PHONY: %-erom-tpm-enable-create
+%-erom-tpm-enable-create:
+	@$(if $(filter $(call hl-device, $@), $(HL_SEC_FIT_CREATE_ARCH)), \
+		${BR2_EXTERNAL}/board/habanalabs/common/create-simple.sh erom_tpm_enable fw-loader-conf no-use-elbi no-use-fwupdate load-bootloader)
+
 .PHONY: %-security-enable-install
 %-security-enable-install:
 	@$(if $(filter $(call hl-device, $@), $(HL_SEC_FIT_CREATE_ARCH)), \
